tests/testthat/test-DataValidation.R

```diff
@@ -0,0 +1,108 @@
+
+test_that("ValidateUploadedData() identifies valid continuous long data", {
+  data <- CleanData(read.csv("Cont_long.csv"))
+  
+  validation_result <- ValidateUploadedData(data, "Continuous")
+  
+  expect_true(validation_result$valid)
+  expect_equal(validation_result$message, "Data is valid")
+})
+
+test_that("ValidateUploadedData() identifies valid continuous wide data", {
+  data <- CleanData(read.csv("Cont_wide.csv"))
+  
+  validation_result <- ValidateUploadedData(data, "Continuous")
+  
+  expect_true(validation_result$valid)
+  expect_equal(validation_result$message, "Data is valid")
+})
+
+test_that("ValidateUploadedData() identifies valid binary long data", {
+  data <- CleanData(read.csv("Binary_long.csv"))
+  
+  validation_result <- ValidateUploadedData(data, "Binary")
+  
+  expect_true(validation_result$valid)
+  expect_equal(validation_result$message, "Data is valid")
+})
+
+test_that("ValidateUploadedData() identifies valid binary wide data", {
+  data <- CleanData(read.csv("Binary_wide.csv"))
+  
+  validation_result <- ValidateUploadedData(data, "Binary")
+  
+  expect_true(validation_result$valid)
+  expect_equal(validation_result$message, "Data is valid")
+})
+
+test_that("ValidateUploadedData() identifies invalid coninuous long data, missing colunm 'N'", {
```

Comment:
coninuous (also in several other places)

Reply:
Fixed